src/commands/multi_block/monitor.rs

```diff
@@ -104,27 +111,43 @@ where
             .ok_or(Error::AccountDoesNotExists)?;
         prometheus::set_balance(account_info.data.free as f64);
 
+        let current_block_round = state.round;
+
+        // This checking logic runs on EVERY block
+        let rounds_to_potentially_clear =
+            collect_past_rounds_to_clear(&submitted_rounds, current_block_round).await;
+
+        check_and_clear_discarded_submissions(
```

Comment:
you could re-use the storage from [the state](https://github.com/paritytech/polkadot-staking-miner/blob/main/src/commands/multi_block/types.rs#L139)  instead of re-fetching inside this function